modin/core/dataframe/base/dataframe/utils.py

```diff
@@ -48,3 +53,71 @@ class JoinType(Enum):  # noqa: PR01
     LEFT = "left"
     RIGHT = "right"
     OUTER = "outer"
+
+
+def join_columns(
+    left: pandas.Index,
+    right: pandas.Index,
+    left_on: Union[Iterable[IndexLabel], IndexLabel],
+    right_on: Union[Iterable[IndexLabel], IndexLabel],
+    suffixes: Tuple[str, str],
+) -> Tuple[pandas.Index, Dict[IndexLabel, IndexLabel], Dict[IndexLabel, IndexLabel]]:
+    """
+    Compute resulting columns for the two dataframes being merged.
+
+    Parameters
+    ----------
+    left : pandas.Index
+        Columns of the left frame to join.
+    right : pandas.Index
+        Columns of the right frame to join.
+    left_on : list-like or scalar
+        Column names on which the frames are joined in the left DataFrame.
+    right_on : list-like or scalar
+        Column names on which the frames are joined in the right DataFrame.
+    suffixes : tuple[str, str]
+        A 2-length sequence containing suffixes to append to the intersected columns.
+
+    Returns
+    -------
+    pandas.Index, dict[IndexLabel -> IndexLabel], dict[IndexLabel -> IndexLabel]
+        Returns columns for the resulting frame and mappings of old to new column
+        names for `left` and `right` accordingly.
+    """
+    left_on = cast(Iterable[IndexLabel], [left_on] if is_scalar(left_on) else left_on)
+    right_on = cast(
+        Iterable[IndexLabel], [right_on] if is_scalar(right_on) else right_on
+    )
+
+    left_conflicts = set(left) & (set(right) - set(right_on))  # type: ignore # set() doesn't understand IndexLabel
+    right_conflicts = set(right) & (set(left) - set(left_on))  # type: ignore # set() doesn't understand IndexLabel
```

Comment:
`set(Iterable[Hashable])` apparently doesn't work for some reason, according to `mypy` set only allows `Iterable[Union[int, str, ... enumeration of scalar types, Tuple[Hashable, ...], None]]`:
```
modin/core/dataframe/base/dataframe/utils.py:96:52: error: Argument 1 to "set" has incompatible type "Iterable[Hashable]"; expected "Iterable[Union[str, bytes, date, timedelta, int, float, complex, Tuple[Hashable, ...], None]]"  [arg-type]
modin/core/dataframe/base/dataframe/utils.py:97:53: error: Argument 1 to "set" has incompatible type "Iterable[Hashable]"; expected "Iterable[Union[str, bytes, date, timedelta, int, float, complex, Tuple[Hashable, ...], None]]"  [arg-type]
Found 2 errors in 1 file (checked 1 source file)
```